client/src/lib/Injector.js

```diff
@@ -28,6 +28,8 @@ class Injector {
    */
   getComponentByDataType(dataType) {
     switch (dataType) {
+      case 'String':
+        return this.components.LabelField;
```

Comment:
String is single-line editable text, not single-line uneditable text. This should default to `TextField`, and Labels should be marked as Structural (SCHEMA_DATA_TYPE_STRUCTURAL).